packages/fetchai/connections/gym/connection.py

```diff
@@ -57,6 +57,7 @@ def __init__(self, address: Address, gym_env: gym.Env):
         self._threaded_pool: ThreadPoolExecutor = ThreadPoolExecutor(
             self.THREAD_POOL_SIZE
         )
+        self.logger: Union[logging.Logger, logging.LoggerAdapter] = logger
```

Comment:
where is logger initiated?

Reply:
right!

Reply:
the logger on the right is the top-level module logger.
It is overwritten by `GymConnection.connect` to replace it with the wrapped logger (i.e. with the agent name prepended).
We can't to that at init time because we create the loggers after the component is instantiated. We could have done the other way, but would have been too intrusive (we would have to modify the init of many components)